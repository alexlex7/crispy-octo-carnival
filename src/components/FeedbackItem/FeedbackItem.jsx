import React from 'react';
import styles from './FeedbackItem.module.css';
import PropTypes from 'prop-types';
import StarRating from '../StarRating/StarRating';

export default function FeedbackItem({ item }) {
  const { firstName, feedback, rating } = item;
  return (
    <li className={styles.item}>
      <p className={styles.name}>{firstName}</p>
      <div className={styles.wrapper}>
        <StarRating rating={rating} isReadOnly={true} />
        <p>{rating}</p>
      </div>
      <p className={styles.text}>{feedback}</p>
    </li>
  );
}

FeedbackItem.propTypes = {
  item: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    feedback: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
  }).isRequired,
};
