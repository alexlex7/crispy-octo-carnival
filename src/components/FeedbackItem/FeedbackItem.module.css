.item {
  padding: 20px 20px 20px 35px;
  width: 380px;
  height: 230px;
  border-radius: 5px;
  box-shadow: rgb(0 0 0 / 20%) 0px 3px 3px -2px,
    rgb(0 0 0 / 14%) 0px 3px 4px 0px, rgb(0 0 0 / 12%) 0px 1px 8px 0px;
}

.wrapper {
  margin-bottom: 10px;
  display: flex;
  align-items: center;
  column-gap: 10px;
}

.name {
  margin-bottom: 10px;

  font-size: 16px;
  font-weight: 700;
}

.text {
  max-height: 100px;
  font-size: 14px;
  font-weight: 400;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 4;
}
