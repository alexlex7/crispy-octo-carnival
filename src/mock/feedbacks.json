[
  {
    "firstName": "Alex",
    "lastName": "Losev",
    "phone": "380986670595",
    "email": "a@t.com",
    "feedback": "The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!",
    "department": "value1",
    "permission": true,
    "rating": 5,
    "id": 7
  },
  {
    "firstName": "Alex",
    "lastName": "Losev",
    "phone": "380986670595",
    "email": "a@t.com",
    "feedback": "The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!",
    "department": "value1",
    "permission": true,
    "rating": 3.5,
    "id": 7
  },
  {
    "firstName": "Alex",
    "lastName": "Losev",
    "phone": "380986670595",
    "email": "a@t.com",
    "feedback": "The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!",
    "department": "value1",
    "permission": true,
    "rating": 4.5,
    "id": 7
  }
]
