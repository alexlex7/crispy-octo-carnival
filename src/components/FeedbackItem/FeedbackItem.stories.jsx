import React from 'react';
import feedbacks from '../../mock/feedbacks.json';
import FeedbackItem from './FeedbackItem';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Feedback/FeedbackItem',
  component: FeedbackItem,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <FeedbackItem {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  item: feedbacks[0],
};
