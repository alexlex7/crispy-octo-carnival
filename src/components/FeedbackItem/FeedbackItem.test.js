import { render, screen } from '@testing-library/react';
import FeedbackItem from './FeedbackItem';
import feedbacks from '../../mock/feedbacks.json';

test('render feedback item', () => {
  const { asFragment } = render(<FeedbackItem item={feedbacks[0]} />);

  const textLink = screen.getByText(feedbacks[0].firstName);
  expect(textLink).toBeInTheDocument();

  const feedbackTextLink = screen.getByText(feedbacks[0].feedback);
  expect(feedbackTextLink).toBeInTheDocument();

  const ratingLink = screen.getByText(feedbacks[0].rating);
  expect(ratingLink).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
