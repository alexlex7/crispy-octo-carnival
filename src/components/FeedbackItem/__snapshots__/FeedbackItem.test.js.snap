// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render feedback item 1`] = `
<DocumentFragment>
  <li
    class="item"
  >
    <p
      class="name"
    >
      Alex
    </p>
    <div
      class="wrapper"
    >
      <div
        class="ratingWrapper"
      >
        <div
          class="starsWrapper"
        >
          <svg
            class="icon"
            fill="none"
          >
            star.svg
          </svg>
          <svg
            class="icon"
            fill="none"
          >
            star.svg
          </svg>
          <svg
            class="icon"
            fill="none"
          >
            star.svg
          </svg>
          <svg
            class="icon"
            fill="none"
          >
            star.svg
          </svg>
          <svg
            class="icon"
            fill="none"
          >
            star.svg
          </svg>
        </div>
        <div
          class="starsActiveWrapper"
          style="width: 100%;"
        >
          <div
            style="display: flex;"
          >
            <svg
              class="activeIcon"
              fill="orange"
            >
              star.svg
            </svg>
            <svg
              class="activeIcon"
              fill="orange"
            >
              star.svg
            </svg>
            <svg
              class="activeIcon"
              fill="orange"
            >
              star.svg
            </svg>
            <svg
              class="activeIcon"
              fill="orange"
            >
              star.svg
            </svg>
            <svg
              class="activeIcon"
              fill="orange"
            >
              star.svg
            </svg>
          </div>
        </div>
      </div>
      <p>
        5
      </p>
    </div>
    <p
      class="text"
    >
      The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!
    </p>
  </li>
</DocumentFragment>
`;
